src/basic/dirent-util.h

```diff
@@ -34,6 +34,10 @@ struct dirent *readdir_no_dot(DIR *dirp);
  * Let's introduce a simple wrapper. */
 #if !HAVE_POSIX_GETDENTS
 static inline ssize_t posix_getdents(int fd, void *buf, size_t nbyte, int flags) {
+        assert(fd >= 0);
+        assert(buf);
+        assert(nbyte > 0);
+        assert(flags == 0); /* Currently flags must be zero. */
```

Comment:
I don't think we should assert on this. This is a compat wrapper, and an actual implementation of posix_getdents() would instead return something like EINVAL/EOPNOTSUPP, which we shall just follow here